### PR TITLE
Host/Domain change support added for local avatar url

### LINKED ISSUE #64: Avatar support when moving hosts / changing domains
**Is your enhancement related to a problem? Please describe.**
[Originally reported on WordPress.org support forum](https://wordpress.org/support/topic/make-urls-immune-to-host-changes/):
> After moving host / changing domain the avatars will still use the OLD URLs. I suppose a nice way to circumvent this were to store the attachment ID instead and get the URLs of the resized versions from that.

**Describe the solution you'd like**
TBD

**Designs**
n/a

**Describe alternatives you've considered**
none

**Additional context**
n/a

### PR DESCRIPTION
**Pull request by jayedul**
### Description of the Change
Current domain/host will be used for local avatar URLs even after site migration.

Closes #64 

### How to test the Change

- Simply edit the image url host in `simple_local_avatar` user meta, maybe in database directly for easy test.
- Reload the page wherever the avatar image is supposed to load
- The image should be regenerated for that specific avatar size and the URL should get back to its original state based on current host/domain. 

### Changelog Entry

> Fixed - Local avatar urls remain old after domain/host change

### Credits

Props @jayedul 

### Checklist:

- [x] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [x] All new and existing tests pass.

### CODE REVIEW

**Comment by ravinderk** [comment]
@jayedul Is this pull request ready for review?

**Comment by jeffpaul** [comment]
@ravinderk let's go ahead and proceed with review on the PR

File: includes/class-simple-local-avatars.php
Comment (ravinderk) [inline includes/class-simple-local-avatars.php:383]: @jayedul  Can we make the following improvements to this function:

- Can we remove `! empty( $local_avatars['media_id'] )` check on line `357` because we already validate it last line `340`?
- We should generate a `full` image URL only if:
   - The `full` image URL does not exist
   - or The `full` image URL does not point to the same site upload directory

**Review by ravinderk — CHANGES_REQUESTED** [review]
@jayedul I left one suggestion. Let me know if you have questions.

**Comment by jeffpaul** [comment]
@faisal-alvi can you help with the PHPUnit failures here as well as general testing/code review to help get this ready for merge/release?

**Comment by faisal-alvi** [comment]
The PHPUnit and Cypress test failures are fixed.

Also, I've tested this in a single setup site, and it's working perfectly. I'd like to move forward with testing it in a multisite network setup before we proceed with the merge. Hopefully, I'll have some time next week to conduct that testing.

**Review by faisal-alvi — APPROVED** [review]
It was tested in a multisite network, working as expected. Thanks for the work, everyone!

**Comment by faisal-alvi** [comment]
Note: Unable to merge, requires one more approval. cc @ravinderk

**Review by ravinderk — APPROVED** [review]

**Comment by ravinderk** [comment]
@faisal-alvi pr approved 🤞

**Pull request merged**
